consensus/src/consensus_observer/observer/consensus_observer.rs

```diff
@@ -817,6 +843,98 @@ impl ConsensusObserver {
         }
     }
 
+    /// Processes the ordered block with window message
+    async fn process_ordered_block_with_window_message(
+        &mut self,
+        peer_network_id: PeerNetworkId,
+        _message_received_time: Instant,
+        ordered_block_with_window: OrderedBlockWithWindow,
+    ) {
+        // If execution pool is disabled, ignore the message
+        let execution_pool_window_size = match self.get_execution_pool_window_size() {
+            Some(window_size) => window_size,
+            None => {
+                // Log the failure and update the invalid message counter
+                warn!(
+                    LogSchema::new(LogEntry::ConsensusObserver).message(&format!(
+                        "Received ordered block with window message but execution pool is disabled! Ignoring: {:?}",
+                        ordered_block_with_window.ordered_block().proof_block_info()
+                    ))
+                );
+                update_invalid_message_counter(
+                    &peer_network_id,
+                    metrics::ORDERED_BLOCK_WITH_WINDOW_LABEL,
+                );
+                return;
+            },
+        };
+
```

Comment:
Can we return early here if `execution_pool_window_size` is 1?

Reply:
Even if it's 1, we still need to process the actual block (below). My next PR will do some of that. 😄